### ISSUE TITLE
A JavaScript error occurred in the main process (Err_Invalid_Callback)

### ISSUE THREAD

**Issue by MrSteve920** [ytmdesktop/ytmdesktop#978]
- [x ] I understand that **YTMDesktop have NO affiliation with Google or YouTube**.
- [x ] I verified that there is no open issue for the same subject to the best of my ability (all the process errors are named the same issue for the most part; I tried to review as many of them as I could.)

**Describe the bug**
Javascript error on app startup (Err_Invalid_Callback). Settings menu does not render properly once error is dismissed.

**To Reproduce**
1. Open application
2. Error message is displayed
3. Settings menu is not rendered properly

**Expected behavior**
No error message on startup. Settings menu loads in values for all setting fields.

**Screenshots**

Error message at startup:
![Capture](https://user-images.githubusercontent.com/41220784/172698394-203cca0b-6b32-4ec8-a4c8-0ce985713672.PNG)

All the settings sub menus look like this:
![Screenshot 2022-06-08 150547](https://user-images.githubusercontent.com/41220784/172698480-5cf1f7f8-afe0-47fc-98c4-923504e7d3a5.png)

**Environment:**
 * YTMDesktop version: ***1.14.2***
 * OS: ***Windows***
 * OS version: ***10***
 * Arch: ***x64***
* Installation way: ***.exe***

**Comment by Alipoodle** [comment]
Settings is already marked as another issue: #967 
This can be fixed manually by downloading the Language you want to use from Locales report and placing it into `%appdata%/youtube-music-desktop-app/locales` and a reload of the settings (or client to fix it everywhere) will fix it.

The callback issue, as I cannot see the line it's failing on, I cannot help more with this as I do not have the full information of the error 😖

**Comment by MrSteve920** [comment]
Do the language files download on startup from some location that may or may not be blocked by a work vpn connection? 😐

I can't really replicate the error fully now.  I rebooted my system and moved from my work vpn to being at home on the internet there. I already tried rebooting at work before submitting this issue but that didn't change anything. After startup at home, the error message is gone and the settings menu labels are fixed with english localization. I can recreate the error popup by going offline, removing the english json, and restarting the app but the whole ui behind the error code is completely missing by using this recreate method which isn't consistent behavior with how this was erroring out previously, not that it really matters at this point.

**Comment by Alipoodle** [comment]
They're hosted on GitHub and accessed via their content site, so 😓 ideally it shouldn't be blocked as it's not like it's a bad website, but could raise some flags.

I'll take look into this a bit more to try and prevent the errors and possibly raise a flag to the user about the missing locale. 
Thanks for the amazing info though, as this helps drastically with fixing the issue!! 🥰

**Comment by MrSteve920** [comment]
Noted issues return when opening the app while on my work vpn. The en locale file is being overwritten with an empty file instead of leaving the existing ones that got downloaded at my home connection or one I downloaded from the locale rep and put in appdata in place each time the app is restarted or the setting menu is accessed. Wherever the files are hosted, for some reason that connection is blocked for me at work. 🙄 Nothing else on github that I've ever run into is blocked though but my IT department never ceases to amaze me. 

It looks like something with the auto updating locales in conjunction with whatever network limitations I have in place is causing this issue. Issue is reproducable back to the nightly build from Feb 12, 2021 but is not present in the nightly build from Jan 26, 2021 or anything prior including the v1.13.0 Sept 7, 2020 release.

**Comment by Alipoodle** [comment]
https://raw.githubusercontent.com/ytmdesktop/ytmdesktop-locales/master/locales/en.json
Is an example of the URL we use 😁 so try manually opening that to see if it fails. It's a GitHub link which is used for accessing the Raw file contents.

**Comment by MrSteve920** [comment]
No failure on my end; that link opens up just fine while "working". Is there a more detailed error log I can pull for you to take a look at?

**Comment by DrSuSE** [comment]
I can confirm on my end that en.json is being overwritten with a blank file upon launch and then the en.json.tmp file is populated with strings that just reference the string name again.  For example:

` "LABEL_SETTINGS_TAB_GENERAL_CUSTOM_THEME": "LABEL_SETTINGS_TAB_GENERAL_CUSTOM_THEME",`

As far as I can tell, it has nothing to do with network or being able to reach/or access any location.  I downloaded the linked en.json file and set it as Read Only as a temporary workaround.  This created a second error.  So now on launch I get the noted javascript error and an error about the Read Only flag on that en.json file.

**Comment by akabakass** [comment]
I've the same issue on ubuntu 22.04, ytmd 1.14.2.
I've noticed that if i open the settings, close them, wait a minute and open them again, everything is back to normal.
ytmd is launched at startup in my case so maybe the internet connection is not yet fully done?

**Comment by Alipoodle** [comment]
Closing issue due to no longer supporting YouTube Music Desktop v1 + Release of YouTube Music v2.
Please re-open this issue if this is still occurring in the latest build of the application, however as a re-write has been done on the entire project most issues will be fixed, no longer a feature or re-worked.
